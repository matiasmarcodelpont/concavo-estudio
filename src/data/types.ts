export type WithNullProp<T, K extends keyof T> = Omit<T, K> & {
  [P in K]: null
}

export interface Reference {
  slug: string
}

export type OmitReferences<T> = {
  [K in keyof T as Exclude<T[K], null> extends Reference | Reference[] ? never : K]: T[K]
}

export interface Producto {
  name: string
  slug: string
  colaborador: Reference | null
}

export type ConcavoProducto = WithNullProp<Producto, 'colaborador'>

export function isConcavoProducto(producto: Producto): producto is ConcavoProducto {
  return producto.colaborador === null
}

export interface Ambiente {
  name: string
  slug: string
  productos: Reference[]
}

interface BaseColaborador {
  name: string
  slug: string
  website: string
}

export type MainColaborador = BaseColaborador & {
  isMain: true
  description: string
  email: string
  address: string
}

export type StandardColaborador = BaseColaborador & {
  isMain: false
  description: string | null
  email: string | null
  address: string | null
}

export type Colaborador = MainColaborador | StandardColaborador

export function isMainColaborador(colaborador: Colaborador): colaborador is MainColaborador {
  return colaborador.isMain
}

export interface DataSet {
  productos: Producto[]
  ambientes: Ambiente[]
  colaboradores: Colaborador[]
}
