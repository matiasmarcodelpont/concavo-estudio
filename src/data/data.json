{
  "rooms": [
    {
      "slug": "living-comedor",
      "name": "Living/Comedor",
      "products": [
        {
          "slug": "lampara-hierro"
        },
        {
          "slug": "piso-madera"
        }
      ],
      "contributors": []
    },
    {
      "slug": "cocina",
      "name": "Cocina",
      "products": [
        {
          "slug": "azulejos-guardados"
        }
      ],
      "contributors": []
    },
    {
      "slug": "lavadero",
      "name": "Lavadero",
      "products": [],
      "contributors": []
    },
    {
      "slug": "escritorio",
      "name": "Escritorio",
      "products": [
        {
          "slug": "lampara-hierro"
        },
        {
          "slug": "piso-madera"
        }
      ],
      "contributors": []
    },
    {
      "slug": "toilet",
      "name": "Toilet",
      "products": [],
      "contributors": []
    },
    {
      "slug": "cuarto-alegra",
      "name": "Cuarto Alegra",
      "products": [],
      "contributors": []
    },
    {
      "slug": "baño-alegra",
      "name": "Baño Alegra",
      "products": [],
      "contributors": []
    },
    {
      "slug": "cuarto-principal",
      "name": "Cuarto principal",
      "products": [],
      "contributors": []
    },
    {
      "slug": "baño-principal",
      "name": "Baño principal",
      "products": [],
      "contributors": []
    },
    {
      "slug": "galeria",
      "name": "Galería",
      "products": [],
      "contributors": []
    },
    {
      "slug": "jardin",
      "name": "Jardín",
      "products": [],
      "contributors": []
    }
  ],
  "products": [
    {
      "slug": "piso-madera",
      "name": "Piso simil Madera",
      "contributor": {
        "slug": "elegance"
      }
    },
    {
      "slug": "lampara-hierro",
      "name": "Lámpara de diseño con detalles de Hierro",
      "contributor": {
        "slug": "luz-viva"
      }
    },
    {
      "slug": "sofa-carlos",
      "name": "Sofá Carlos V",
      "contributor": {
        "slug": "muebles-arte"
      }
    },
    {
      "slug": "azulejos-guardados",
      "name": "Azulejos Guardados",
      "contributor": {
        "slug": "cocina-design"
      }
    },
    {
      "slug": "bacha-8080",
      "name": "Bacha 8080",
      "contributor": null
    },
    {
      "slug": "bañadera-10001",
      "name": "Bañadera 10001",
      "contributor": null
    }
  ],
  "contributors": [
    {
      "slug": "elegance",
      "name": "Elegance",
      "website": "www.pisoselegance.es",
      "isMain": true,
      "description": "Líder en revestimientos de suelos elegantes y duraderos para hogares y negocios. Calidad, innovación y compromiso con la sostenibilidad nos distinguen.",
      "email": "info@pisoselegance.com",
      "address": "Avenida del Diseño 123, Ciudad Elegante, España"
    },
    {
      "slug": "luz-viva",
      "name": "Luz Viva",
      "website": "www.luzviva.com",
      "isMain": true,
      "description": "Líder en sistemas de iluminación innovadores y eficientes para hogares y negocios. Calidad, diseño y compromiso con la sostenibilidad nos distinguen.",
      "email": "info@luzviva.com",
      "address": "Calle de la Luz 456, Ciudad Brillante, México"
    },
    {
      "slug": "muebles-arte",
      "name": "Muebles de Arte",
      "website": "www.mueblesdearte.com",
      "isMain": false,
      "description": null,
      "email": null,
      "address": "Calle del Diseño 789, Ciudad Creativa, Argentina"
    },
    {
      "slug": "cocina-design",
      "name": "Cocina Design",
      "website": "www.cocinadesign.com",
      "isMain": false,
      "description": "Especialistas en diseño y equipamiento de cocinas de alta gama. Ofrecemos soluciones personalizadas para crear la cocina de tus sueños, combinando estilo, funcionalidad y calidad.",
      "email": null,
      "address": null
    }
  ]
}
