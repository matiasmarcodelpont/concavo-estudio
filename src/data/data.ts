import { DataSet } from './types'

const data: DataSet = {
  ambientes: [
    {
      slug: 'living-comedor',
      name: 'Living/Comedor',
      productos: [
        {
          slug: 'lampara-hierro',
        },
        {
          slug: 'piso-madera',
        },
      ],
      images: [
        {
          src: '0.jpeg',
          width: 1500,
          height: 1080,
          puntitos: [{ productoSlug: 'azulejos-guardados', coordinates: { x: 760, y: 430 } }],
        },
        { src: '1.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'cocina',
      name: 'Cocina',
      productos: [
        {
          slug: 'azulejos-guardados',
        },
      ],
      images: [
        { src: '0.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'lavadero',
      name: 'Lavadero',
      productos: [],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    // {
    //   slug: 'escritorio',
    //   name: 'Escritorio',
    //   productos: [
    //     {
    //       slug: 'lampara-hierro',
    //     },
    //     {
    //       slug: 'piso-madera',
    //     },
    //   ],
    // },
    // {
    //   slug: 'toilet',
    //   name: 'Toilet',
    //   productos: [],
    // },
    // {
    //   slug: 'cuarto-alegra',
    //   name: 'Cuarto Alegra',
    //   productos: [],
    // },
    // {
    //   slug: 'baño-alegra',
    //   name: 'Baño Alegra',
    //   productos: [],
    // },
    // {
    //   slug: 'cuarto-principal',
    //   name: 'Cuarto principal',
    //   productos: [],
    // },
    // {
    //   slug: 'baño-principal',
    //   name: 'Baño principal',
    //   productos: [],
    // },
    // {
    //   slug: 'galeria',
    //   name: 'Galería',
    //   productos: [],
    // },
    // {
    //   slug: 'jardin',
    //   name: 'Jardín',
    //   productos: [],
    // },
  ],
  productos: [
    {
      slug: 'piso-madera',
      name: 'Piso simil Madera',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      link: 'https://example.com',
      colaborador: {
        slug: 'elegance',
      },
    },
    {
      slug: 'lampara-hierro',
      name: 'Lámpara de Hierro',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      link: 'https://example.com',
      colaborador: {
        slug: 'luz-viva',
      },
    },
    {
      slug: 'sofa-carlos',
      name: 'Sofá Carlos V',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      link: 'https://example.com',
      colaborador: {
        slug: 'muebles-arte',
      },
    },
    {
      slug: 'azulejos-guardados',
      name: 'Azulejos Guardados',
      link: 'https://example.com',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',

      colaborador: {
        slug: 'cocina-design',
      },
    },
    {
      slug: 'bacha-costa',
      name: 'Bacha Costa',
      link: 'https://example.com',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      colaborador: null,
    },
    {
      slug: 'bañadera-kahlo',
      name: 'Bañadera Kahlo',
      link: 'https://example.com',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      colaborador: null,
    },
    {
      slug: 'bañadera-kahlo',
      name: 'Bañadera Kahlo',
      link: 'https://example.com',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      colaborador: null,
    },
    {
      slug: 'bañadera-kahlo',
      name: 'Bañadera Kahlo',
      link: 'https://example.com',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Incidunt voluptas explicabo aliquam tenetur. Eligendi deleniti ratione doloribus dolores dolorum, reprehenderit maiores odit itaque quos tenetur, architecto earum est, iste accusantium.',
      colaborador: null,
    },
  ],
  colaboradores: [
    {
      slug: 'elegance',
      name: 'Elegance',
      contact: 'Contacto: 5473-1862',
    },
    {
      slug: 'luz-viva',
      name: 'Luz Viva',
      contact: 'Contacto: 5473-1862',
    },
    {
      slug: 'muebles-arte',
      name: 'Muebles de Arte',
      contact: 'Contacto: 5473-1862',
    },
    {
      slug: 'cocina-design',
      name: 'Cocina Design',
      contact: 'Contacto: 5473-1862',
    },
  ],
}

export default data
