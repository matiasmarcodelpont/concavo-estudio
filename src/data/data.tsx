import { DataSet } from './types'

const data: DataSet = {
  ambientes: [
    {
      slug: 'living-comedor',
      name: 'Living/Comedor',
      productos: [{ slug: 'travertino' }, { slug: 'pietra' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'cocina',
      name: 'Cocina',
      productos: [{ slug: 'travertino' }, { slug: 'pietra' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'lavadero',
      name: 'Lavadero',
      productos: [{ slug: 'travertino' }, { slug: 'pietra' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'escritorio',
      name: 'Escritorio',
      productos: [{ slug: 'travertino' }, { slug: 'pietra' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'toilet',
      name: 'Toilet',
      productos: [{ slug: 'travertino' }, { slug: 'inodoro-ap' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'cuarto-alegra',
      name: 'Cuarto Alegra',
      productos: [{ slug: 'travertino' }, { slug: 'pietra' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'baño-alegra',
      name: 'Baño Alegra',
      productos: [{ slug: 'travertino' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'cuarto-principal',
      name: 'Cuarto principal',
      productos: [{ slug: 'travertino' }, { slug: 'pietra' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'baño-principal',
      name: 'Baño principal',
      productos: [{ slug: 'travertino' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'galeria',
      name: 'Galería',
      productos: [{ slug: 'travertino' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
    {
      slug: 'jardin',
      name: 'Jardín',
      productos: [{ slug: 'travertino' }, { slug: 'teclas-de-luz' }],
      images: [
        { src: '0.jpeg', width: 1500, height: 2000, puntitos: [] },
        { src: '1.jpeg', width: 1500, height: 1080, puntitos: [] },
        { src: '2.jpeg', width: 1500, height: 1080, puntitos: [] },
      ],
    },
  ],
  productos: [
    {
      slug: 'travertino',
      name: 'Travertino',
      description: (
        <>
          Travertino clasico al agua.
          <br />
          medida: 40.5 x Largo Libre
          <br />
          apomazado semi taponado
        </>
      ),
      link: 'https://destefano1913.com/tienda/baldosa/travertino-clasico-al-agua-baldosa/',
      colaborador: { slug: 'de-stefano' },
    },
    // {
    //   slug: 'griferia-monocomando-gr',
    //   name: 'Grifería Monocomando - Línea GR',
    //   description: (
    //     <>
    //       Su tecnología SOFT FLOW mantiene las salpicaduras al mínimo
    //       <br />
    //       El disco cerámico, un componente de válvula central, está cubierto con una capa especial de DLC (Carbono de
    //       Diamante). Esto asegura una calidad duradera sobre la grifería convencional.
    //     </>
    //   ),
    //   link: 'https://destefano1913.com/?s=griferia+TOTO+GR&post_type=product&type_aws=true',
    //   colaborador: { slug: 'toto' },
    // },
    {
      slug: 'inodoro-ap',
      name: 'INODORO DE COLGAR AP + WASHLET C5 ELONGATED ',
      description: (
        <>
          Los Washlet de TOTO®️ son asientos con bidet electrónico incorporado, climatización del agua y de la
          tabla,secador automático y desodorizante.
          <br />
          Cuentan con tecnología CEFIONTEC®️, cerámica ultra lisa que no deja lugar para residuos. TORNADO FLUSH®️, el
          sistema de descarga 360°, elimina por completo todos los desechos utilizando menos agua.
        </>
      ),
      link: 'https://destefano1913.com/tienda/sanitario/toto-inodoro-ap-ct426-con-washlet-c5-cisterna-tece-mamposteria/',
      colaborador: { slug: 'toto' },
    },
    {
      slug: 'pietra',
      name: 'PIETRA',
      description: (
        <>
          COLOR: BLANCO MALLORCA
          <br />
          Revestimiento decorativo acrílico, de acabado similar a la piedra natural. Producto elástico y resistente a la
          intemperie que aplicado en alto espesor protege y decora, sustituyendo al sistema tradicional de revoque fino
          más pintura. Indicado para interiores y exteriores tanto en obras nuevas como en refacciones.
        </>
      ),
      link: 'https://www.sinteplast.com.ar/arquitectonica/revestimientos/pietra-sinteplast',
      colaborador: { slug: 'sinteplast' },
    },
    {
      slug: 'teclas-de-luz',
      name: 'TECLAS DE LUZ',
      description: (
        <>
          Interruptor inteligente
          <br />
          Color: Blanco
        </>
      ),
      link: 'https://www.akaienergy.com.ar/ficha.php?id=64',
      colaborador: { slug: 'akai' },
    },
    // {
    //   slug: 'iluminacion',
    //   name: 'ILUMINACION',
    //   description: (
    //     <>
    //       iluminacion profesional/ iluminacion general del hogar
    //       <br />
    //       Aplique: FELTRE color blanco
    //       <br />
    //       Lampara: LED MR16 - 1180
    //       <br />
    //       Tiras LED: NODOT 8400 cob
    //     </>
    //   ),
    //   link: 'https://italighting.com.ar/img/productos/ITAlighting-Catalogo.pdf',
    //   colaborador: { slug: 'akai' },
    // },
    // {
    //   slug: 'instalaciones-sanitarias',
    //   name: 'instalaciones sanitarias',
    //   description: (
    //     <>Elegimos Sigas para tendido de gas, Hidro 3 para tendido de agua fria y caliente y Duratop para cloacal</>
    //   ),
    //   link: '',
    //   colaborador: { slug: 'resta' },
    // },
    // {
    //   slug: 'ladrillo-klimablock',
    //   name: 'ladrillo klimablock',
    //   description: (
    //     <>
    //       Optamos por poner ladrillo hueco Klimablock en todas las paredes exteriores. Un ladrillo de 27cm con óptima
    //       aislación de la humedad, térmica, acústica y al fuego.
    //     </>
    //   ),
    //   link: '',
    //   colaborador: { slug: 'cisilotto' },
    // },
    {
      slug: 'griferia',
      name: 'GRIFERIA',
      description: (
        <>
          Resta distribuye
          <br />
          Marca: Peirano
          <br />
          Modelo: Vera{' '}
        </>
      ),
      link: '',
      colaborador: { slug: 'resta' },
    },
    {
      slug: 'inodoro',
      name: 'INODORO',
      description: (
        <>
          Marca: Roca.
          <br />
          Modelo: The Gap
        </>
      ),
      link: '',
      colaborador: { slug: 'resta' },
    },
    {
      slug: 'inodoro-2',
      name: 'INODORO',
      description: (
        <>
          Marca: Roca
          <br />
          Modelo: Inspira
        </>
      ),
      link: '',
      colaborador: { slug: 'resta' },
    },
  ],
  colaboradores: [
    {
      slug: 'de-stefano',
      name: 'De Stefano',
      level: 'black',
      whatsapp: '11 4991-8163',
      instagram: '@destefano1913',
      email: 'marketing@destefano.com.ar',
    },
    {
      slug: 'toto',
      name: 'Toto',
      level: 'black',
      whatsapp: '11 4991-8163',
      instagram: '@toto.argentina',
      email: 'marketing@destefano.com.ar',
    },
    {
      slug: 'sinteplast',
      name: 'Sinteplast',
      level: 'black',
      website: 'https://www.sinteplast.com.ar/contacto/',
    },
    {
      slug: 'resta',
      name: 'Resta',
      level: 'black',
      whatsapp: '11 3697-7275',
      email: 'axelmontenegro@sanitariosresta.com.ar',
      instagram: '@restadesign.it',
    },
    {
      slug: 'akai',
      name: 'Grupo Akai (Akai + Italighting)',
      level: 'black',
      reference: 'Contacto Comercial: Sebastian Zazzarro',
      email: 'sebastian@italighting.com.ar',
    },
    {
      slug: 'cristalizando',
      name: 'Cristalizando',
      level: 'gold',
      whatsapp: '(0387) 443-643',
      instagram: '@cristalizandosa',
      email: 'ventas@cristalizando.com',
    },
    {
      slug: 'muchtek',
      name: 'Muchtek Tecnoperfiles Group',
      level: 'silver',
      email: 'perfiles@muchtek.com',
      instagram: '@muchtek.tecnoperfilesgroup',
    },
    {
      slug: 'topcret',
      name: 'Topcret',
      level: 'silver',
      whatsapp: '11 6688-8128',
      instagram: '@topcretargentina',
      email: 'argentina@topcret.com',
    },
    {
      slug: 'cisilotto',
      name: 'Cisilotto',
      level: 'standard',
      whatsapp: '11 3857-1212',
      instagram: '@cisilotto.hnos',
      email: 'ventas@cisilotto.com.ar',
    },
    {
      slug: 'ecoaislacion',
      name: 'Ecoaislacion',
      level: 'standard',
      whatsapp: '11 2696-6442',
      instagram: '@ecoaislacion.oeste',
    },
    {
      slug: 'diclimatizacion',
      name: 'Diclimatizacion',
      level: 'standard',
      whatsapp: '11 3216-7203',
      instagram: '@diclimatizacion',
      reference: 'Luciano Marchi',
    },
  ],
}

export default data
