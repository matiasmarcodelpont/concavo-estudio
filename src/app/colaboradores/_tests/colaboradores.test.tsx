import '@testing-library/jest-dom'
import { render, screen, within } from '@testing-library/react'

import Colaboradores from '../page'
import { DataSet } from '@/data/types'

jest.mock(
  '@/data/data.ts',
  () =>
    ({
      ambientes: [],
      productos: [],
      colaboradores: [
        {
          slug: 'luz-viva',
          name: 'Luz Viva',
          level: 'black',
        },
        {
          slug: 'cocina-design',
          name: 'Cocina Design',
          level: 'black',
        },
        {
          slug: 'elegance',
          name: 'Elegance',
          level: 'black',
        },
        {
          slug: 'muebles-arte',
          name: 'Muebles de Arte',
          level: 'black',
        },
      ],
    }) satisfies DataSet,
)

describe('Colaboradores', () => {
  it('renders Colaboradores unchanged', () => {
    const { container } = render(<Colaboradores />)
    expect(container).toMatchSnapshot()
  })

  it('renders Colaboradores name', () => {
    render(<Colaboradores />)

    const heading = screen.getByRole('heading', { name: 'Colaboradores' })
    expect(heading).toBeInTheDocument()
  })

  it('renders all Colaboradores', () => {
    render(<Colaboradores />)

    const ColaboradoresList = screen.getByRole('list')
    expect(ColaboradoresList).toBeInTheDocument()

    const colaboradores = within(ColaboradoresList).getAllByRole('listitem')
    expect(colaboradores).toHaveLength(4)
  })
})
