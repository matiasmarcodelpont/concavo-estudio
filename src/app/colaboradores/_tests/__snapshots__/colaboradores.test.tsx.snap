// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colaboradores renders Colaboradores unchanged 1`] = `
<div>
  <main
    class="text-center"
  >
    <section
      aria-labelledby="colaboradores-heading"
      class="mx-6 sm:mx-10 md:mx-12 mb-12"
    >
      <span
        style="font-size: 0px;"
      />
      <h1
        class="font-editorial-new tracking-[0.01em] font-light uppercase mt-12 mb-6 sm:mt-16 sm:mb-9 md:mt-20 md:mb-12 text-3xl md:text-4xl opacity-0"
        id="colaboradores-heading"
      >
        Colaboradores
      </h1>
      <ul
        aria-label="Colaboradores"
        class="flex flex-wrap gap-10 sm:gap-12 md:gap-14 lg:gap-16 mb-10 sm:mb-12 md:mb-14 lg:mb-16 justify-center"
      >
        <li
          class="list-none"
        >
          <span
            style="font-size: 0px;"
          />
          <div
            class="w-[200px] sm:w-[300px] h-[40px] m-auto relative mb-3 transition-all duration-1000 opacity-0"
          >
            <img
              alt="Luz Viva"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=640&q=75 640w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=750&q=75 750w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=828&q=75 828w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=1080&q=75 1080w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=1200&q=75 1200w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=1920&q=75 1920w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=2048&q=75 2048w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </li>
        <li
          class="list-none"
        >
          <span
            style="font-size: 0px;"
          />
          <div
            class="w-[200px] sm:w-[300px] h-[40px] m-auto relative mb-3 transition-all duration-1000 opacity-0"
          >
            <img
              alt="Cocina Design"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=640&q=75 640w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=750&q=75 750w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=828&q=75 828w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=1080&q=75 1080w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=1200&q=75 1200w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=1920&q=75 1920w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=2048&q=75 2048w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </li>
        <li
          class="list-none"
        >
          <span
            style="font-size: 0px;"
          />
          <div
            class="w-[200px] sm:w-[300px] h-[40px] m-auto relative mb-3 transition-all duration-1000 opacity-0"
          >
            <img
              alt="Elegance"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fcolaboradores%2Felegance.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fcolaboradores%2Felegance.png&w=640&q=75 640w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=750&q=75 750w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=828&q=75 828w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=1080&q=75 1080w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=1200&q=75 1200w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=1920&q=75 1920w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=2048&q=75 2048w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </li>
        <li
          class="list-none"
        >
          <span
            style="font-size: 0px;"
          />
          <div
            class="w-[200px] sm:w-[300px] h-[40px] m-auto relative mb-3 transition-all duration-1000 opacity-0"
          >
            <img
              alt="Muebles de Arte"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=640&q=75 640w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=750&q=75 750w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=828&q=75 828w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=1080&q=75 1080w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=1200&q=75 1200w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=1920&q=75 1920w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=2048&q=75 2048w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
